src/cst.hpp

```diff
@@ -34,19 +37,67 @@ struct DeclarationData {
 	}
 };
 
+using FuncArguments = std::vector<DeclarationData>;
+
 struct Declaration : public CST {
+	// This function is very cold -- it's ok to use virtuals
+	virtual InternedString const& identifier_virtual() const = 0;
```

Comment:
Doubt it will ever be a bottleneck, and since this is for parsing purposes it kinda does not matter.